docs/plugins/pluggable_rule_engine.md

```diff
@@ -32,29 +32,32 @@ Because the rule engine plugins are just one of many iRODS plugin types, the REP
 
-If there is more than one matching rule for a particular PEP, the first one loaded will fire first.  If it succeeds, then the others are ignored.  If the first one fails, then the next matching rule is fired.  If it fails, then the framework will continue to "fall through" until there are no more matching rules.  The return code of the last matching rule will be the one that is returned.
+If there is more than one matching rule for a particular PEP, only the first one loaded will fire.  If it succeeds, then the others are ignored.  If the first one fails, then an error is returned to the client and processing stops.
 
-![Rule Engine Plugin Framework Diagram](../images/rule_engine_plugin_framework_diagram.jpg)
+The REPF can be instructed to do different things based on the code returned from a rule. See the following table:
 
-This "fall through" mechanism also applies across rule engine plugins (meaning, across language as well).
+| Symbolic Code Name         | Code      | Behavior |
+| -------------------------- | --------- | -------- |
+| RULE_ENGINE_CONTINUE       | 5000000   | Allow the next **rule engine plugin** to process the PEP. |
+| RULE_ENGINE_SKIP_OPERATION | 5001000   | Do not execute the server operation following the **pre** PEP. This can only be returned from a **pre** PEP. If returned from a PEP that isn't a **pre** PEP, the server will log a warning and ignore the instruction. |
+| SYS_NOT_SUPPORTED          | -66000    | Synonymous with RULE_ENGINE_CONTINUE. |
 
-There are three scopes where the "fall through" will occur:
+`RULE_ENGINE_CONTINUE` is important for implementing maintainable policy.  It allows rule code to be split across rule engine plugins. Doing this allows policy implementers to separate concerns and focus on smaller chunks of code.  This is demonstrated in the example below.
 
-- within the framework itself, falling from one plugin to another
-- within a plugin, falling from one rulebase to another
-- within a rulebase, falling from one rule to another
-
-This nested scope defines the order in which identically named rules take precedent on a particular iRODS server.
+`RULE_ENGINE_SKIP_OPERATION` allows policy implementers to skip or replace the behavior of an API operation.  Care must be taken when using this instruction.  The Consortium highly recommends avoiding it unless absolutely necessary.
 
 ### Example
 
-Consider the following `server_config.json` with the accompanying `example.re`, `core.re` and `core.py` rulebases:
+Consider the following `server_config.json` with the accompanying `core.re` and `core.py`:
+
+!!! Note
+    `core.py` is normally located in the same directory as `server_config.json`. See the [Python Rule Engine Plugin](https://github.com/irods/irods_rule_engine_plugin_python) for additional details.
```

Comment:
Removing this. I thought this would be helpful to people unfamiliar with the PREP (i.e. it isn't obvious what core.py is until they install the plugin), but I don't think we need to think about that right now.

If someone is confused by this, they will let us know and we can adjust things then.